src/scripts/main.js

```diff
@@ -1,3 +1,32 @@
 'use strict';
 
-// write code here
+const tree = document.querySelector('.tree');
+const lis = tree.querySelectorAll('li');
+
+lis.forEach((el) => {
+  const text = el.childNodes[0].nodeValue.trim();
+  const span = document.createElement('span');
+
+  span.textContent = text;
+  el.childNodes[0].replaceWith(span);
+});
+
+tree.addEventListener('click', (ev) => {
+  ev.stopPropagation();
+
+  const span = ev.target.closest('span');
+
+  if (!span) {
+    return;
+  }
+
+  const ul = span.nextSibling;
```

Comment:
The current logic assumes that the next sibling of the `span` is always the `ul` element. This might not be accurate if there are other elements or text nodes between the `span` and `ul`. Consider using `nextElementSibling` instead of `nextSibling` to ensure you are targeting the correct `ul` element.